flake.nix

```diff
@@ -10,13 +10,41 @@
         # TODO: We have parts of our CI that rely on clang-format-15, but that is only avalible on github:nixos/nixpkgs/nixos-unstable
         llvmPkgs = pkgs.llvmPackages_14;
         pythonEnv = import ./nix/pyenv.nix { pkgs = pkgs; };
-        openssl_0_9_8 = import ./nix/openssl_0_9_8.nix { pkgs = pkgs; };
-        openssl_1_0_2 = import ./nix/openssl_1_0_2.nix { pkgs = pkgs; };
         openssl_1_1_1 = import ./nix/openssl_1_1_1.nix { pkgs = pkgs; };
-        openssl_3_0 = import ./nix/openssl_3_0.nix { pkgs = pkgs; };
-        libressl = import ./nix/libressl.nix { pkgs = pkgs; };
         corretto-8 = import nix/amazon-corretto-8.nix { pkgs = pkgs; };
         gnutls-3-7 = import nix/gnutls.nix { pkgs = pkgs; };
+        libcrypto_packages = [
+          # Libcryptos we always expect to be available
+          openssl_1_1_1
+          gnutls-3-7
+        ];
```

Comment:
nice! much easier to know what we are using. can you add a comment why we need `gnutls`. Is that libcrypto or `libssl`.. if so maybe it can live in common_packages?

Reply:
We're using the client and server utilities for interop testing: [gnutls-serv](https://github.com/aws/s2n-tls/blob/main/tests/integrationv2/providers.py#L860) and [gnutls-client](https://github.com/aws/s2n-tls/blob/main/tests/integrationv2/providers.py#L826). Maybe the variable needs a better name, tls_interop_pacakges, tls_packages, ?

Reply:
`tls_packages ` seems more self-explanatory.